src/app/Controllers/friendrequests.component.ts

```diff
@@ -16,7 +17,8 @@ export class FriendRequestsComponent implements OnInit, OnDestroy {
     constructor(
         private friendsApiService: FriendsApiService,
         public cacheService: CacheService,
-        private headerService: HeaderService
+        private headerService: HeaderService,
+        public router: Router
```

Comment:
Is this router necessary?

Reply:
It's used in the html template. I think it can be removed but also have to change the html file.